lib/samples/animate_3d_graphic/animate_3d_graphic.dart

```diff
@@ -0,0 +1,647 @@
+                      setModalState(() => _cameraHeading = value);
+                      setState(
+                        () => _cameraController.cameraHeadingOffset = value,
+                      );
+                    },
+                  ),
+                  _buildSlider(
+                    label: 'Pitch Offset',
+                    value: _cameraPitch,
+                    min: 0,
+                    max: 180,
+                    onChanged: (value) {
+                      setModalState(() => _cameraPitch = value);
+                      setState(
+                        () => _cameraController.cameraPitchOffset = value,
+                      );
+                    },
+                  ),
+                  SwitchListTile(
+                    title: const Text('Auto Heading'),
+                    value: _autoHeading,
+                    onChanged: (value) {
+                      setModalState(() => _autoHeading = value);
+                      setState(
+                        () => _cameraController.isAutoHeadingEnabled = value,
+                      );
+                    },
+                  ),
+                  SwitchListTile(
+                    title: const Text('Auto Pitch'),
+                    value: _autoPitch,
+                    onChanged: (value) {
+                      setModalState(() => _autoPitch = value);
+                      setState(
+                        () => _cameraController.isAutoPitchEnabled = value,
+                      );
+                    },
+                  ),
+                  SwitchListTile(
+                    title: const Text('Auto Roll'),
+                    value: _autoRoll,
+                    onChanged: (value) {
+                      setModalState(() => _autoRoll = value);
+                      setState(
+                        () => _cameraController.isAutoRollEnabled = value,
+                      );
+                    },
+                  ),
+                ],
+              ),
+            );
+          },
+        );
+      },
+    );
+  }
+
+  // Builds a labeled slider widget.
+  Widget _buildSlider({
+    required String label,
+    required double value,
+    required double min,
+    required double max,
+    required ValueChanged<double> onChanged,
+  }) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text('$label: ${value.toStringAsFixed(0)}'),
+        Slider(
+          value: value,
+          min: min,
+          max: max,
+          divisions: (max - min).toInt(),
+          label: value.toStringAsFixed(0),
+          onChanged: onChanged,
+        ),
+        const SizedBox(height: 8),
+      ],
+    );
+  }
+}
+
+// An enum of the different mission selections available in this sample.
+enum Mission { grandCanyon, hawaii, pyrenees, snowdon }
+
+// To provide labels and ArcGIS Online item IDs for each mission.
+extension MissionLabel on Mission {
```

Comment:
I don't think you need to use an `extension` here -- just add these methods to the body of the `enum Mission` declaration above.